pkg/scheduler/cache/node_info.go

```diff
@@ -256,172 +264,76 @@ func (r *Resource) SetMaxResource(rl v1.ResourceList) {
 // the returned object.
 func NewNodeInfo(pods ...*v1.Pod) *NodeInfo {
 	ni := &NodeInfo{
-		requestedResource:   &Resource{},
-		nonzeroRequest:      &Resource{},
-		allocatableResource: &Resource{},
-		TransientInfo:       newTransientSchedulerInfo(),
-		generation:          nextGeneration(),
-		usedPorts:           make(util.HostPortInfo),
-		imageStates:         make(map[string]*ImageStateSummary),
+		RequestedResource:   &Resource{},
+		NonzeroRequest:      &Resource{},
+		AllocatableResource: &Resource{},
+		TransientInfo:       NewTransientSchedulerInfo(),
+		Generation:          nextGeneration(),
+		UsedPorts:           make(util.HostPortInfo),
+		ImageStates:         make(map[string]*ImageStateSummary),
 	}
 	for _, pod := range pods {
 		ni.AddPod(pod)
 	}
 	return ni
 }
 
-// Node returns overall information about this node.
-func (n *NodeInfo) Node() *v1.Node {
```

Comment:
We should not modify the `NodeInfo` API in this PR. Also, by replacing functions like this with a direct access, we risk introducing nil panics into other components.